### PR TITLE
refactor: replace _apacheCommonsCodec Base64 with JDK Base64 and enhance error handling (#801)

### LINKED ISSUE #801: Remove classes from _apacheCommonsCodec package
**What needs to be done? Please describe.**

The package `org.operaton.bpm.engine.impl.digest._apacheCommonsCodec` contains classes `Base64` and `StringUtils` which likely can be replaced by Java API. `StringUtils` is only used by `Base64`.

The target is to remove the package by equivalent changes.

Identify usages of `Base64` and analyze if core Java API is a valid replacement.

For example, `org.operaton.bpm.client.interceptor.auth.BasicAuthProvider#encodeToBase64` uses this:
```
    return Base64.getEncoder()
      .encodeToString(stringAsBytes);
```

Ideally cover methods that make use of the methods that should be removed by unit tests, e.g. provide a unit test for `org.operaton.bpm.engine.impl.digest.Base64EncodedHashDigest#encodeHash` and then refactor.

### PR DESCRIPTION
**Pull request by ungerts**
### Summary

This PR removes the custom `_apacheCommonsCodec` Base64 implementation and replaces it with the standard `java.util.Base64` API.

### Related Issues

Closes #801

### Motivation and Context

The custom Base64 and StringUtils classes in the `_apacheCommonsCodec` package were redundant and could be replaced by Java's built-in functionality. Additionally, external input (e.g., HTTP headers) may contain invalid Base64, which necessitates proper exception handling and clear logging to maintain robustness and transparency.

### Implementation Details

- Replaced `org.operaton.bpm.engine.impl.digest._apacheCommonsCodec.Base64` with `java.util.Base64`.
- Removed the obsolete `_apacheCommonsCodec` package.
- Updated `HttpBasicAuthenticationProvider` to catch decoding exceptions and log a warning if Base64 input is invalid.
- Logging in the Rest module now follows the standard pattern used in the engine module.
- `RestLogger` was marked as a sealed class to enforce that all loggers extend `BaseLogger`, ensuring consistent logging structure.
- Added a new test class, `HttpBasicAuthenticationProviderTest`, to verify decoding behavior and edge cases.

### Screenshots or Logs (if applicable)

N/A

### Breaking Changes?

- [ ] Yes
- [x] No

---

#### Administrative
- [x] I have read and followed the [contribution guidelines](https://github.com/operaton/operaton/blob/main/CONTRIBUTING.md)
- [x] I confirm that my contribution complies with the [Apache License 2.0](https://www.apache.org/licenses/LICENSE-2.0)
- [x] This PR references an issue (e.g., `Fixes #123`)

### Change Type

Please select the type of change this PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional or API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### CODE REVIEW

File: engine-rest/engine-rest/src/test/java/org/operaton/bpm/engine/rest/security/auth/impl/HttpBasicAuthenticationProviderTest.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  private static final String CREDENTIALS =
+      Base64.getEncoder().encodeToString(String.format("%s:%s", USER_ID, PASSWORD).getBytes());
+
+  private static final String AUTHORIZATION_HEADER = BASIC_AUTH_HEADER_PREFIX + CREDENTIALS;
+
+  @Test
+  void testExtractAuthenticatedUserNoAuthHeader() {
+    HttpBasicAuthenticationProvider provider = new HttpBasicAuthenticationProvider();
+    HttpServletRequest request = Mockito.mock(HttpServletRequest.class);
+    ProcessEngine engine = Mockito.mock(ProcessEngine.class);
```
Comment (kthoms) [inline engine-rest/engine-rest/src/test/java/org/operaton/bpm/engine/rest/security/auth/impl/HttpBasicAuthenticationProviderTest.java:31]: Maybe set up provider, request, engine as fields and initialize by a BeforeEach method? These three are always initialized the same in several methods.

File: engine-rest/engine-rest/src/test/java/org/operaton/bpm/engine/rest/standalone/AbstractAuthenticationFilterTest.java [hunk 2]
```diff
@@ -84,7 +83,7 @@ public void setup() {
 
     // example method
     ProcessDefinition mockDefinition = MockProvider.createMockDefinition();
-    List<ProcessDefinition> mockDefinitions = Arrays.asList(mockDefinition);
+    List<ProcessDefinition> mockDefinitions = Collections.singletonList(mockDefinition);
```
Comment (kthoms) [inline engine-rest/engine-rest/src/test/java/org/operaton/bpm/engine/rest/standalone/AbstractAuthenticationFilterTest.java:86]: nit: I would prefer List.of() nowadays, but also fine

File: engine-rest/engine-rest/src/main/java/org/operaton/bpm/engine/rest/impl/AuthLogger.java [hunk 3]
```diff
@@ -1,9 +1,11 @@
 /*
+ * Copyright 2025 the Operaton contributors.
+ *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
+ * You may obtain a copy of the License at:
  *
- *     https://www.apache.org/licenses/LICENSE-2.0
+ *     http://www.apache.org/licenses/LICENSE-2.0
```
Comment (kthoms) [inline engine-rest/engine-rest/src/main/java/org/operaton/bpm/engine/rest/impl/AuthLogger.java:8]: We should use HTTPS Urls. Since I have added it wrong to the Contributing guide and there are a lot of duplications, I have opened https://github.com/operaton/operaton/issues/852 to change that.
No need to do something here now.

**Review by kthoms — APPROVED** [review]
Thanks, looking good now.
